base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,31 +21,246 @@
+
+    if (ComboBox_GetCount(hCombo) == 0)
+    {
+        WCHAR sShow[100];
+        WCHAR sHide[100];
+        WCHAR sInactive[100];
+        LoadStringW(NULL, IDS_NOTIF_BEH_SHOW, sShow, _countof(sShow));
+        LoadStringW(NULL, IDS_NOTIF_BEH_HIDE, sHide, _countof(sHide));
+        LoadStringW(NULL, IDS_NOTIF_BEH_HIDE_INACTIVE, sInactive, _countof(sInactive));
+        ComboBox_AddString(hCombo, sShow);
+        ComboBox_AddString(hCombo, sHide);
+        ComboBox_AddString(hCombo, sInactive);
+        ComboBox_SetCurSel(hCombo, 0);
+    }
+
+    int iItem = itemdata->iItem;
+    if (iItem > -1)
+    {
+        POINT pt;
+        RECT rc;
+        ListView_GetItemPosition(hListView, iItem, &pt);
+        GetWindowRect(hListView, &rc);
+        POINT lv = { rc.left, rc.top };
+        ScreenToClient(hwnd, &lv);
+
+        LVITEM lvItem;
+        lvItem.iItem = iItem;
+        lvItem.iSubItem = 0;
+        lvItem.mask = LVIF_PARAM;
+        ListView_GetItem(hListView, &lvItem);
+
+        ComboBox_SetCurSel(hCombo, lvItem.lParam);
+
+        int iNameWidth = ListView_GetColumnWidth(hListView, 0);
+        int iBehaviorWidth = ListView_GetColumnWidth(hListView, 1);
+
+        ComboBox_SetItemData(hCombo, 0, iItem);
+        SetWindowPos(hCombo, HWND_TOP, iNameWidth + lv.x, pt.y + lv.y, iBehaviorWidth, 20, SWP_SHOWWINDOW);
+    }
+    else
+    {
+        ShowWindow(hCombo, SW_HIDE);
+    }
+}
+
+VOID SetItemBehavior(HWND hwnd, HWND hCombo)
+{
+    int iState = ComboBox_GetCurSel(hCombo);
+    int iItem = (int)ComboBox_GetItemData(hCombo, 0);
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+
+    WCHAR szBehavior[100];
+    int resId = iState == 0 ? IDS_NOTIF_BEH_SHOW : (iState == 1 ? IDS_NOTIF_BEH_HIDE : IDS_NOTIF_BEH_HIDE_INACTIVE);
+    LoadStringW(NULL, resId, szBehavior, _countof(szBehavior));
+
+    LVITEM lvItem;
+    ZeroMemory(&lvItem, sizeof(lvItem));
+    lvItem.mask = LVIF_TEXT;
+    lvItem.iItem = iItem;
+    lvItem.iSubItem = 1;
+    lvItem.cchTextMax = 100;
+    lvItem.pszText = szBehavior;
+    ListView_SetItem(hListView, &lvItem);
+
+    ZeroMemory(&lvItem, sizeof(lvItem));
+    lvItem.mask = LVIF_PARAM;
+    lvItem.iSubItem = 0;
+    lvItem.lParam = iState;
+    lvItem.iItem = iItem;
+    ListView_SetItem(hListView, &lvItem);
+
+    ShowWindow(hCombo, SW_HIDE);
+}
+
+VOID ResetBehaviors(HWND hwnd)
+{
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+    int count = ListView_GetItemCount(hListView);
+    WCHAR szBehavior[100];
+    LoadStringW(NULL, IDS_NOTIF_BEH_HIDE_INACTIVE, szBehavior, _countof(szBehavior));
+
+    LVITEM lviBehavior;
+    ZeroMemory(&lviBehavior, sizeof(lviBehavior));
+    lviBehavior.mask = LVIF_TEXT;
+    lviBehavior.iSubItem = 1;
+    lviBehavior.cchTextMax = 100;
+    lviBehavior.pszText = szBehavior;
+
+    LVITEM lviName;
+    ZeroMemory(&lviName, sizeof(lviName));
+    lviName.mask = LVIF_PARAM;
+    lviName.iSubItem = 0;
+    lviName.lParam = 2;
```

Comment:
It should actually be OK to simplify these two things using a single LVITEM structure.
You zero it out as you do, then you set the members that do not change (lParam, cchTextMax and pszText), then, in the for loop and before you call each ListView_SetItem, you set (for the first one), iItem to i, iSubItem to 0 and mask to LVIF_PARAM, then (for the second one) you keep the iItem, you change iSubItem to 1 and mask to LVIF_TEXT.